mempool/src/core_mempool/mempool.rs

```diff
@@ -178,8 +199,12 @@ impl Mempool {
             let account_sequence_number = self.sequence_number_cache.get(&txn.address);
             let seen_previous = seq > 0 && seen.contains(&(txn.address, seq - 1));
             // include transaction if it's "next" for given account or
-            // we've already sent its ancestor to Consensus
-            if seen_previous || account_sequence_number == Some(&seq) {
+            // we've already sent its ancestor to Consensus. In the case of CRSNs, we can safely
+            // assume that it can be included.
+            if seen_previous
+                || account_sequence_number == Some(&seq)
+                || matches!(txn.seqno_type, AccountSequenceNumber::CRSN { .. })
```

Comment:
What happens if the CRSN on a tx is (e.g.) 20, but a subsequent transaction force expires to shift the LHS of window past 20? Will the tx be evicted elsewhere, or is it possible that it will be included in the block here (and subsequently discarded)?

Reply:
The tx would be evicted the next time a transaction a transaction is sent -- we only get knowledge about the LHS of the window at the time of submission of the transaction. So it could be possible that it would be included in the same block as the force-expiration transaction, or a different block before the next transaction from the account is sent.